### ISSUE TITLE
Integrate the virtualizarr dmr++ parser into pydap

### ISSUE THREAD

**Issue by ayushnag** [pydap/pydap#417]
### Description

There have been some discussions to have the `virtualizarr` dmrpp parser as a part of`pydap`. This would be beneficial since the parser can make sure to maintain strict adherence to DAP4.0 protocol and have future support by OPeNDAP

Here is the existing `virtualizarr` parser's [source code for reference](https://github.com/zarr-developers/VirtualiZarr/blob/main/virtualizarr/readers/dmrpp.py)

One factor to consider is that the parser has `virtualizarr` as a dependency since `dmrpp.py` needs some classes/functions from `virtualizarr` such as the `ChunkManifest` and `ZArray` classes. I think there are two options to resolve this:
1) Add `virtualizarr` as a (optional?) dependency to `pydap`
- This would be the easiest way to get this code merged.
- `virtualizarr.open_virtual_dataset` would just call `pydap.dmrpp.parse_dataset()` for example
2) Allow `pydap` to natively support dmr++ and create a dataset with chunk information
- Basically allow pydap to read a dmrpp and output a `Dataset` object with chunk information, compression information, etc.
- Then virtualizarr can call `pydap.Dataset(file)` and shuffle the information (vars, attributes, chunks) into xarray's data model.
- This option makes more sense to me but will require some updates and design decisions to be made in pydap (how the chunk manifest is represented)

cc. @Mikejmnez, @jgallagher59701, @betolink, @TomNicholas

**Comment by Mikejmnez** [comment]
Yes -  I agree with you that option 2, while it may take bit of time, is probably the best course of action. I too have been slowly expanding pydap's dmr parser and pydap's data model, to cover all of DAP4, and so it is best to have a single source of truth dmr parser. 

Thanks for the good work @ayushnag !

**Comment by TomNicholas** [comment]
I don't know much about the details of the pydap side but I also like the idea of working towards having a single source of truth parser, and that parser living in the pydap repo so that IIUC the stewards of the protocol are also maintainers of the parser.
